File: Frontend/src/features/columnMappings/types/columnMapping.ts
export interface ColumnMapping {
  id: string;
  sourceColumn: string;
  targetProperty: string;
  targetDisplayName: string;
  dataType: string;
  isRequired: boolean;
  isActive: boolean;
  displayOrder: number;
  createdOn: string;
  modifiedOn?: string | null;
}

export interface ColumnMappingFormValues {
  sourceColumn: string;
  targetProperty: string;
  targetDisplayName: string;
  dataType: string;
  isRequired: boolean;
  isActive: boolean;
  displayOrder: number;
}

export interface ColumnValueMapping {
  id: string;
  targetProperty: string;
  sourceValue: string;
  targetValue: string;
  isActive: boolean;
  createdOn: string;
  modifiedOn?: string | null;
}

export interface ColumnValueMappingFormValues {
  targetProperty: string;
  sourceValue: string;
  targetValue: string;
  isActive: boolean;
}

File: Models/ColumnMapping.cs
using System.ComponentModel.DataAnnotations;

namespace HRMS.Api.Models
{
    public class ColumnMapping
    {
        [Key]
        public Guid Id { get; set; }

        [MaxLength(200)]
        public string SourceColumn { get; set; } = string.Empty;

        [MaxLength(200)]
        public string TargetProperty { get; set; } = string.Empty;

        [MaxLength(200)]
        public string TargetDisplayName { get; set; } = string.Empty;

        [MaxLength(50)]
        public string DataType { get; set; } = "string";

        [MaxLength(100)]
        public string EntityType { get; set; } = "employee-import";

        public bool IsRequired { get; set; }

        public bool IsActive { get; set; } = true;

        public int DisplayOrder { get; set; }

        public DateTime CreatedOn { get; set; }

        public DateTime? ModifiedOn { get; set; }
    }
}
